src/merkle/store/mod.rs

```diff
@@ -257,6 +257,22 @@ impl MerkleStore {
         Ok(tree_depth)
     }
 
+    // DATA EXTRACTORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns a subset of this Merkle store such that the returned Merkle store contains all
+    /// nodes which are descendants of the specified roots.
+    ///
+    /// The roots for which no descendants exist in this Merkle store are ignored.
```

Comment:
Why do we silently ignore roots with no descendants?  Is there a reason why we don't error?

Reply:
I had two reasons in mind:

1. We don't really impose any shape on the output. A single root could have a huge subtree under it, or it could be a minimal subtree of depth 1 - so, the jump to say that it could be absent entirely is not that big in my mind.
2. This should simplify working with the account storage. We could pass an iterator over set of all non-zero slot values into this, and it will extract all relevant subtrees.

I did realize that it might be better to change the parameter to be an iterator - so, I updated the method signature.

Reply:
Makes sense - thanks for the explanation. 

> 2. This should simplify working with the account storage. We could pass an iterator over set of all non-zero slot values into this, and it will extract all relevant subtrees.

I think we should also be able to pass a single root (root of the account storage slots) and it should extract the slots tree and any nested merkle structures?

Reply:
> I think we should also be able to pass a single root (root of the account storage slots) and it should extract the slots tree and any nested merkle structures?

If we do that, the cloned subset would contain the depth 8 tree for the slots tree as well. I guess one of the questions is how do we get the slots `SimpleSmt` out of the Merkle store of a transaction. We might need to add another method to the `MerkleStore` for this - maybe something like `MerkleStore::get_tree(root, depth)`.